linkerd/main/src/main/scala/io/buoyant/linkerd/Main.scala

```diff
@@ -64,10 +64,16 @@ object Main extends App {
   private def initAdmin(
     config: Linker.LinkerConfig,
     linker: Linker
-  ): Closable with Awaitable[Unit] = {
+  ): Seq[Closable with Awaitable[Unit]] = {
     val server = linker.admin.serve(this, LinkerdAdmin(config, linker))
     log.info(s"serving ${linker.admin.scheme} admin on ${server.boundAddress}")
-    server
+
+    config.admin.flatMap(_.httpIdentifierPort) match {
+      case Some(p) =>
+        val idServer = linker.admin.serveHandler(new InetSocketAddress(p.port), LinkerdAdmin.identifierHandler(config, linker))
```

Comment:
The issue with putting it on the existing admin port, I think, is that it would have to be served at a particular URL.  But the URL can factor into identification.